scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1654,124 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if type(shape) != tuple:
+        # in case shape is not passed in as a tuple
+        shape = (shape,)
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+         
+
+@pytest.mark.parametrize("dtype", DTYPES)
+def test_pttrf_pttrs(dtype):
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250*np.finfo(dtype).eps
+    atol = 100*np.finfo(dtype).eps
+    # n is the length diagonal of A
+    n = 10
+    # create diagonals according to size and dtype
+    if dtype in REAL_DTYPES:
+        # add 2 so that the matrix will be diagonally dominant
+        d = generate_random_dtype_array(n, dtype) + 2 
+    else:
+        # diagonal d should always be real.
+        # for complex add 4 so it will be dominant
+        d = generate_random_dtype_array(n, DTYPES[1]) + 4
+    # diagonal e may be real or complex.
```

Comment:
@mdhaber

Is there still an issue with this way of creating the d diagonal for complex types? It passes through pttrf but with info = 2, so I assume that there is still something wrong with the eigenvalues. Hopefully there's not something trivial that I overlooked.

Reply:
For the complex case, it could be due to the same reason as the NAG example fails. I'll confirm it tomorrow if that's the case.